### DISCUSSION TITLE
[Bug] fnos installs 1panel. After booting, the service is running, but there is no port. 1panel needs to be restarted.

### DISCUSSION THREAD

**Discussion by chengoak** [1Panel-dev/1Panel#7344] [Q&A]
### 1Panel Version

v1.10.21-lts

### Problem Description

开机后，服务运行中，但是没有端口，浏览器访问也没有
![Snipaste_2024-12-12_13-12-05](https://github.com/user-attachments/assets/478859bf-8b8c-4904-a65b-0e905cdbfc7e)

### Steps to Reproduce

重启服务后，端口开放，可访问
![Snipaste_2024-12-12_13-14-11](https://github.com/user-attachments/assets/7ad47c2b-4799-4908-89a6-9e2c7a005f2e)

### Related log output

没有异常日志信息
![Snipaste_2024-12-12_13-30-51](https://github.com/user-attachments/assets/ea347a54-04fe-4270-8fa4-ee487b80588c)

_No response_

**Comment by wanghe-fit2cloud** [comment]
已将该 issue 转为 discussion，看看是否有其他社区用户遇到类似问题，一起讨论解决。
> 1Panel 官方暂未适配 fnOS，目前无法提供该问题的具体解决方案。

**Reply by chengoak** [reply]
请问还有哪里的日志可以排查吗？

**Reply by wanghe-fit2cloud** [reply]
可以参考 Linux 服务器上的  `journalctl -u 1panel -exf` 命令来获取日志。

**Reply by chengoak** [reply]
日志显示是正常启动，尴尬

**Comment by chengoak** [comment]
用AI写了一个脚本，1分钟检测一次

```shell
#!/bin/bash

# 定义端口号和服务名称
PORT=1890
SERVICE_NAME="1panel"

# 检查端口是否在监听
check_port() {
    netstat -tuln | grep ":$PORT " > /dev/null
    return $?
}

# 重启服务
restart_service() {
    echo "端口 $PORT 未被监听，正在重启 $SERVICE_NAME 服务..."
    systemctl restart $SERVICE_NAME
    echo "$SERVICE_NAME 服务已重启。"
}

# 主逻辑
if check_port; then
    echo "端口 $PORT 正在监听，无需操作。"
else
    restart_service
fi
```
